api/config/config.go

```diff
@@ -23,9 +39,38 @@ func init() {
 // DefaultConfig defines the default configuration options for api
 func DefaultConfig() Config {
 	return Config{
-		StartGrpcServer: defaultStartGRPCServer, // note: all bool flags default to false so don't set one of these to true here
-		GrpcServerPort:  defaultGRPCServerPort,
-		StartJSONServer: defaultStartJSONServer,
-		JSONServerPort:  defaultJSONServerPort,
+		StartGrpcServer:    defaultStartGRPCServer, // note: all bool flags default to false so don't set one of these to true here
+		StartGrpcServices:  nil,                    // note: cannot configure an array as a const
+		GrpcServerPort:     defaultGRPCServerPort,
+		NewGrpcServerPort:  defaultNewGRPCServerPort,
+		StartJSONServer:    defaultStartJSONServer,
+		StartNewJSONServer: defaultStartNewJSONServer,
+		JSONServerPort:     defaultJSONServerPort,
+		NewJSONServerPort:  defaultNewJSONServerPort,
+		StartNodeService:   defaultStartNodeService,
+		StartMeshService:   defaultStartMeshService,
+	}
+}
+
+// ParseServicesList enables the requested services
+func (s *Config) ParseServicesList() error {
+	// Make sure all enabled GRPC services are known
+	for _, svc := range s.StartGrpcServices {
+		switch svc {
+		case "mesh":
+			s.StartMeshService = true
+		case "node":
+			s.StartNodeService = true
+		default:
+			return errors.New("unrecognized GRPC service requested: " + svc)
+		}
+	}
+
+	// If JSON gateway server is enabled, make sure at least one
+	// GRPC service is also enabled
+	if s.StartNewJSONServer && !s.StartNodeService {
```

Comment:
what about `StartMeshService `, if this is active then json can server can be started right?

Reply:
Well spotted! Fixed this